src/clj/harja/palvelin/raportointi/raportit/ilmoitukset.clj

```diff
@@ -0,0 +1,200 @@
+        TUR (filtteri-paalla-str (contains? tyypit :tiedoitus))
+        URK (filtteri-paalla-str (contains? tyypit :kysely))
+
+        aloituskuittauksen-ajankohta (cond
+                                       (= aloituskuittauksen-ajankohta :kaikki)
+                                       "Älä rajoita aloituskuittauksella"
+                                       (= aloituskuittauksen-ajankohta :alle-tunti)
+                                       "Alle tunnin kuluessa"
+                                       (= aloituskuittauksen-ajankohta :myohemmin)
+                                       "Yli tunnin päästä")
+
+        myohassa (filtteri-paalla-str (contains? vaikutukset :myohassa))
+        aiheutti-toimenpiteita (filtteri-paalla-str (contains? vaikutukset :aiheutti-toimenpiteita))
+
+        rivit (into []
+                    [(filtterit-rivi "Tiedotettu urakkaan" "Toimenpiteet aloitettu" nil nil true)
+                     (filtterit-rivi valitetty-urakkaan-vakioaikavali toimenpiteet-aloitettu-vakioaikavali nil nil false)
+
+                     (filtterit-rivi "Hakusana" "Selite" "Tienumero" "Tunniste" true)
+                     (filtterit-rivi hakuehto selite tr-numero tunniste false)
+
+                     (filtterit-rivi "Ilmoittaja" "Ilmoittajan puh" nil nil true)
+                     (filtterit-rivi ilmoittaja-nimi ilmoittaja-puhelin nil nil false)
+
+                     (filtterit-rivi "Kuittaamaton" "Vastaanotettu" "Aloitettu" "Lopetettu" true)
+                     (filtterit-rivi kuittaamaton vastaanotettu aloitettu lopetettu false)
+
+                     (filtterit-rivi "TPP" "TUR" "URK" nil true)
+                     (filtterit-rivi TPP TUR URK nil false)
+
+                     (filtterit-rivi "Myöhästyneet ilmoitukset" "Toimenpiteitä aiheuttaneet" "Aloituskuittaus annettu" nil true)
+                     (filtterit-rivi myohassa aiheutti-toimenpiteita aloituskuittauksen-ajankohta nil false)])]
+
+    [:taulukko {:piilota-border? true
+                :sheet-nimi sheet-nimi
+                :hoitokausi-arvotaulukko? false
+                :viimeinen-rivi-yhteenveto? false}
+     (rivi
+      {:otsikko "Käytetyt filtterit" :otsikkorivi-luokka "otsikko-ei-taustaa" :leveys 12 :tyyppi :varillinen-teksti}
+      {:otsikko " " :otsikkorivi-luokka "otsikko-ei-taustaa" :leveys 15 :tyyppi :varillinen-teksti}
+      {:otsikko " " :otsikkorivi-luokka "otsikko-ei-taustaa" :leveys 15 :tyyppi :varillinen-teksti}
+      {:otsikko " " :otsikkorivi-luokka "otsikko-ei-taustaa" :leveys 15 :tyyppi :varillinen-teksti}) rivit]))
+
+(defn suorita [_ _ {:keys [tiedot urakka hallintayksikko filtterit]}]
+  
+  (let [otsikot ["Urakka" "Saapunut"
+                 "Tyyppi" "Selite"
+                 "Lisätieto" "Tie"
+                 "Tila" "Toimenpiteet aloitettu"]
+
+        valittu-ely (try (get ely/elynumero->nimi (Long/parseLong (:elynumero hallintayksikko))) (catch Throwable _ nil))
```

Comment:
try-catchin sijasta käytetään yleensä `(when (:elynumero hallintayksikko) (:elynumero hallintayksikko)) ` tms.
Liian aktiivinen try-catch, erityisesti näin laajalla catchilla voi ottaa joissain tilanteissa kiinni sellaisia tilanteita, jotka on oikeasti virheitä, jolloin oikea virhe jää helposti huomaamatta. Tässä tilanteessa luultavasti ei niin kävisi, mutta silti on hyvä välttää try-catchia kun voi.